### PR TITLE
feat: send free limit reached to PostHog

### PR DESCRIPTION
**Pull request by ShubhamPalriwala**
## What does this PR do?
Sends Free Limit Reached event to Posthog with the property defining if its `inAppSurvey` or `userTargeting` during the sync methods and is wrapped inside an `unstable_cache()` to run only once in 15 days!

### Required

- [ ] Filled out the "How to test" section in this PR
- [ ] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [ ] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

File: apps/web/app/api/v1/client/[environmentId]/in-app/sync/route.ts [hunk 1]
```diff
@@ -56,6 +59,28 @@ export async function GET(
       const currentResponseCount = await getMonthlyTeamResponseCount(team.id);
       isInAppSurveyLimitReached =
         !hasInAppSurveySubscription && currentResponseCount >= PRICING_APPSURVEYS_FREE_RESPONSES;
+      if (isInAppSurveyLimitReached) {
+        await unstable_cache(
+          async () => {
+            const teamDetails = await getTeamDetails(environmentId);
+            if (teamDetails?.teamOwnerId) {
+              console.log("free limit reached");
```
Comment (mattinannt) [inline apps/web/app/api/v1/client/[environmentId]/in-app/sync/route.ts:67]: please remove log

File: apps/web/app/api/v1/client/[environmentId]/in-app/sync/route.ts [hunk 2]
```diff
@@ -56,6 +59,28 @@ export async function GET(
       const currentResponseCount = await getMonthlyTeamResponseCount(team.id);
       isInAppSurveyLimitReached =
         !hasInAppSurveySubscription && currentResponseCount >= PRICING_APPSURVEYS_FREE_RESPONSES;
+      if (isInAppSurveyLimitReached) {
```
Comment (mattinannt) [inline apps/web/app/api/v1/client/[environmentId]/in-app/sync/route.ts:62]: As we have duplicated code here between the endpoints I think it makes sense to outsource this to a separate lib file `apps/web/app/api/v1/client/[environmentId]/in-app/sync/lib/posthog.ts`

This also makes the code in the sync endpoints more readable

**Review by mattinannt — CHANGES_REQUESTED** [review]
@ShubhamPalriwala thanks for the PR; looks pretty good already :-) 

As the logic got a bit bigger, let's outsource it into a route specific lib file 😊

**Comment by ShubhamPalriwala** [comment]
Done, please take a look now

**Review by mattinannt — APPROVED** [review]
@ShubhamPalriwala thanks a lot for the changes; looks great 😊💪

**Pull request merged**